ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/interceptor/AuthorizationInterceptor.scala

```diff
@@ -76,33 +79,43 @@ final class AuthorizationInterceptor(
   private[this] def resolveAuthenticatedUserRights(claimSet: ClaimSet): Future[ClaimSet] =
     claimSet match {
       case ClaimSet.AuthenticatedUser(userIdStr, participantId, expiration) =>
-        Ref.UserId.fromString(userIdStr) match {
-          case Left(err) =>
+        userManagementStoreO match {
+          case None =>
             Future.failed(
-              errorFactories.invalidArgument(None)(s"token $err")(errorLogger)
+              LedgerApiErrors.AuthorizationChecks.Unauthenticated
+                .UserBasedAuthenticationIsDisabled()(errorLogger)
+                .asGrpcError
             )
-          case Right(userId) =>
-            userManagementStore
-              .listUserRights(userId)
-              .flatMap {
-                case Left(msg) =>
-                  Future.failed(
-                    errorFactories.permissionDenied(
-                      s"Could not resolve rights for user '$userId' due to '$msg'"
-                    )(errorLogger)
-                  )
-                case Right(userClaims) =>
-                  Future.successful(
-                    ClaimSet.Claims(
-                      claims = userClaims.view.map(userRightToClaim).toList.prepended(ClaimPublic),
-                      ledgerId = None,
-                      participantId = participantId,
-                      applicationId = Some(userId),
-                      expiration = expiration,
-                      resolvedFromUser = true,
-                    )
-                  )
-              }
+          case Some(userManagementStore) =>
+            Ref.UserId.fromString(userIdStr) match {
+              case Left(err) =>
+                Future.failed(
+                  errorFactories.invalidArgument(None)(s"token $err")(errorLogger)
+                )
+              case Right(userId) =>
+                userManagementStore
+                  .listUserRights(userId)
+                  .flatMap {
+                    case Left(msg) =>
+                      Future.failed(
+                        errorFactories.permissionDenied(
+                          s"Could not resolve rights for user '$userId' due to '$msg'"
+                        )(errorLogger)
+                      )
+                    case Right(userClaims) =>
+                      Future.successful(
+                        ClaimSet.Claims(
+                          claims =
+                            userClaims.view.map(userRightToClaim).toList.prepended(ClaimPublic),
+                          ledgerId = None,
+                          participantId = participantId,
+                          applicationId = Some(userId),
+                          expiration = expiration,
+                          resolvedFromUser = true,
+                        )
+                      )
+                  }
+            }
         }
```

Comment:
This amount of nesting made it a bit difficult to understand what's going on here. Would it be possible to split this into a few smaller methods?

Reply:
Ok